### PR TITLE
Full test coverage for `/pkg/morserino_com/list_com.go`

### PR DESCRIPTION
**Pull request by jmMeessen**
Based on the discussion with Julien, I was able to successfully implement a mock for the `go.bug.st/serial/enumerator/GetDetailedPortsList()` function.

This function scans available port on the system and tries to identify a port used by the Morserino USB serial device.

The same function is used in the `lsiten_com.go` (when auto identification of the port is requested). Note that this file will be substantially refactored in order to render it testable.

### CODE REVIEW

File: pkg/morserino_com/list_com_test.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+// Mock that found no ports
+type mockEnumeratePorts_noPort struct{}
+
+// Mocked method that found no ports
+func (e mockEnumeratePorts_noPort) GetDetailedPortsList() ([]*enumerator.PortDetails, error) {
+	emptyList := []*enumerator.PortDetails{}
+
+	return emptyList, nil
+}
```
Comment (jlevesy) [inline pkg/morserino_com/list_com_test.go]: Even if what you are doing is perfectly valid I'm going to expose another approach.

Here you are actually I implementing one mock per behaviour you want to simulate. What you could do instead is define a parametrized mock, and let the set setup actually configure your mock implementation to simulate your case.

Which would give the following.

```golang
type mockPortEnumerator struct {
        ports []*enumarator.PortDetails
        err error
}

func (e mockPortEnumerator) GetDetailedPortsList() ([]*enumerator.PortDetails, error) {
        return e.ports, e.err
}
```

Then you could replace `mockEnumeratePorts_OK` with

```golang
// ...
portEnumerator := mockPortEnumerator{
    ports: []*enumerator.PortDetails{
        {/* Value 1*/},
        {/* Value 2 */},
    },
}
```

And `mockEnumeratePorts_error` with

```golang
// ...
portEnumerator := mockPortEnumerator{
   err: errors.New("Something went wrong"), 
}
```

Does it makes sense for you ?

File: pkg/morserino_com/list_com_test.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+		Name:         "Morserino Port",
+		IsUSB:        true,
+		VID:          "10C4",
+		PID:          "EA60",
+		SerialNumber: "002",
+	}
+	happyCaseList = append(happyCaseList, &item1)
+	happyCaseList = append(happyCaseList, &item2)
+
+	return happyCaseList, nil
```
Comment (jlevesy) [inline pkg/morserino_com/list_com_test.go]:
```suggestion
	return []*enumerator.PortDetails{
		{
			Name:         "COM Port Name",
			IsUSB:        true,
			VID:          "XXXX",
			PID:          "YYYY",
			SerialNumber: "001",
		},
		{
			Name:         "Morserino Port",
			IsUSB:        true,
			VID:          "10C4",
			PID:          "EA60",
			SerialNumber: "002",
		},
	}, nil
```

Style nit: you can declare initialized slices using this syntax. No need to use `append`

File: pkg/morserino_com/list_com_test.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+		},
+	}
+
+	//validate results
+	if assert.NoError(t, err) {
+		assert.Equal(t, comList.nbrOfPorts, 2)
+		assert.Equal(t, comList.nbrOfMorserinoPorts, 1)
+		assert.Equal(t, comList.morserinoPortName, "Morserino Port")
+		assert.Equal(t, comList.portList, targetPortList)
+	}
```
Comment (jlevesy) [inline pkg/morserino_com/list_com_test.go]:
```suggestion
	require.NoError(t, err)
	
	assert.Equal(t, comList.nbrOfPorts, 2)
	assert.Equal(t, comList.nbrOfMorserinoPorts, 1)
	assert.Equal(t, comList.morserinoPortName, "Morserino Port")
	assert.Equal(t, comList.portList, targetPortList)
```

You can use `github.com/stretchr/testify/require` to stop the test execution if your assertion fails. It saves you  an indentation level here.

**Review by jlevesy — COMMENTED** [review]
:eyes: 

Looks good @jmMeessen !

File: pkg/morserino_com/list_com_test.go (hunk 3 above)
Comment (jmMeessen) [reply pkg/morserino_com/list_com_test.go]: Cool. Learned something new

File: pkg/morserino_com/list_com_test.go (hunk 2 above)
Comment (jmMeessen) [reply pkg/morserino_com/list_com_test.go]: Learned something new. It is indeed more elegant.

File: pkg/morserino_com/list_com_test.go (hunk 1 above)
Comment (jmMeessen) [reply pkg/morserino_com/list_com_test.go]: It keeps the test data nearer to the test. Learned a new trick. Thanks a lot.

**Comment by jmMeessen** [comment]
Thanks a lot for the very instructive comments @jlevesy . Much appreciated.

**Pull request merged**